proxy/src/control/discovery.rs

```diff
@@ -591,14 +550,14 @@ impl<T> DestinationSet<T>
                         "Destination.Get stream ended for {:?}, must reconnect",
                         auth
                     );
-                    return (DestinationServiceQuery::NeedsReconnect, exists);
+                    return (Remote::NeedsReconnect, exists);
                 },
                 Ok(Async::NotReady) => {
-                    return (DestinationServiceQuery::ConnectedOrConnecting { rx }, exists);
+                    return (Remote::connected(rx), exists);
```

Comment:
`connected()` is too misleading since the connection might not be connected yet. I suggest we just use `ConnectedOrConnecting` directly to make it clearer.